tests/suites/test_suite_psa_crypto_util.data

```diff
@@ -0,0 +1,71 @@
+ECDSA Raw -> DER, 192bit, Success
```

Comment:
I understand it makes sense to test with a small curve, and that's what we do in a lot of places already, but we are ]planning on removing this curve](https://github.com/Mbed-TLS/mbedtls/issues/8136) in 4.0 (coming in 2024 unless things go wrong), so it would be better for those tests to use a curve we're going to keep. IMO secp256r1 is the natural choice as it's probably the most widely used curve these days.

Reply:
Sure, I can improve this!

Side question: I picked 192 bit key because, as you might have noticed, I wrote those values manually so I just picked the shortest one. Those integers don't really belong to any real curve. Is that fine or should I use some real coordinate instead? It shouldn't make any difference because we're not doing anything useful with those values a part from conversion...

Reply:
We don't need to test valid signatures. But we do need to have a good coverage of realistic sizes. In particular, given that the DER encoding has a threshold when a part is more than 128 bytes long, we need to have test cases with a 512-bit curve and with a 521-bit curve.

We also need to have test cases where the two integers are shorter in DER (because the raw forms have leading zeros), e.g. a 512-bit curve where the integers nonetheless fit in 128 bytes. If they must be valid signatures, you have to try generating them at random until one has the desired shape, but since these conversion functions don't require valid signatures, you can just create the numbers manually.